test/observer/permissions.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:collection';
+
+import 'package:flutter/services.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+
+import '../test.dart';
+
+/// An observer for platform permission requests.
+class PermissionsChannelObserver {
+  /// The method channel used by the flutter permissions package
+  static const MethodChannel _channel = MethodChannel('flutter.baseflow.com/permissions/methods');
+  final List<Permission> _requestedPermissions = [];  // The permissions which were requested.
+  List<Permission> get requestedPermissions  => UnmodifiableListView(_requestedPermissions);
+  final List<Permission> _checkedPermissions = [];  // The permissions which were checked.
+  List<Permission> get checkedPermissions => UnmodifiableListView(_checkedPermissions);
+  int _openSettingsRequests = 0;  /// The amount of attempts to open the permission settings.
+  int get openSettingsRequests => _openSettingsRequests;
+
+  /// Whether an attempt to open the settings should succeed.
+  bool isOpeningSettingsSuccessful = true;
+  
+  /// Permissions whose values are specified, all other permissions are implicitly granted.
+  final Map<Permission, Future<PermissionStatus> Function()> _specifiedPermissions = {};
+
+  /// Create a new permissions observer, which automatically
+  /// unregisters any previously created observer.
+  PermissionsChannelObserver(TestWidgetsFlutterBinding binding) {
+    binding.defaultBinaryMessenger.setMockMethodCallHandler(_channel, (call) async {
```

Comment:
Please refactor this (and other places as such) to just 

```dart
const MethodChannel('flutter.baseflow.com/permissions/methods').setMockMethodCallHandler((call) async {
// ...
});
```

Reply:
No, this way is deprecated and [was removed in Flutter 2.5](https://docs.flutter.dev/release/breaking-changes/mock-platform-channels), the it is currently is the new [recommended way](https://stackoverflow.com/a/68240542).

Reply:
I'm on Flutter 2.10.3 and is still working for me, in fact, there are places in the app that use this syntax. I would not take an SO answer as a "recommended way". While the current comment on this method says that it should be deprecated, we should not care until it really is deprecated - maybe it will not ever be deprecated.

Reply:
It's true, but the only reason why this still works is because of `TestMethodChannelExtension`, which is defined in `fultter_test/lib/src/deprecated.dart` and whose documentation clearly advertises against it's usage.

Reply:
OK, this makes sense. Let's refactor everything else to use the new binder methods then.